src/content/developers/docs/gas/index.md

```diff
@@ -21,88 +20,113 @@ Since each Ethereum transaction requires computational resources to execute, eac
 
-In essence, gas fees are paid in Ethereum's native currency, ether (ETH). Gas prices are denoted in gwei, which itself is a denomination of ETH - each gwei is equal to 0.000000001 ETH (10<sup>-9</sup> ETH). For example, instead of saying that your gas costs 0.000000001 ether, you can say your gas costs 1 gwei.
+Gas fees are paid in Ethereum's native currency, ether (ETH). Gas prices are denoted in gwei, which itself is a denomination of ETH - each gwei is equal to 0.000000001 ETH (10<sup>-9</sup> ETH). For example, instead of saying that your gas costs 0.000000001 ether, you can say your gas costs 1 gwei.
 
-Let's say Alice has to pay Bob 1ETH.
+## Prior to the London upgrade {#pre-london}
+
+The way transaction fees on the Ethereum network are calculated changed during the London network upgrade. Here is a recap of how things used to work:
+
+Let's say Alice had to pay Bob 1 ETH.
 In the transaction the gas limit is 21,000 units and the gas price is 200 gwei.
 
-Total fee will be: `Gas units (limit) * Gas price per unit`
+Total fee would have been: `Gas units (limit) * Gas price per unit`
 i.e `21,000 * 200 = 4,200,000 gwei` or 0.0042 ETH
 
-Now, when Alice sends the money, 1.0042 ETH will be deducted from Alice's account.
-Bob will be credited 1.0000 ETH.
-Miner gets 0.0042 ETH.
+When Alice sent the money, 1.0042 ETH would be deducted from Alice's account.
+Bob would be credited 1.0000 ETH.
+Miner would receive 0.0042 ETH.
 
 This video offers a concise overview of gas and why it exists:
 
 <iframe width="100%" height="315" src="https://www.youtube.com/embed/AJvzNICwcwc" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
-## Why do gas fees exist? {#why-do-gas-fees-exist}
+## After the London Upgrade {#post-london}
 
-In short, gas fees help keep the Ethereum network secure. By requiring a fee for every computation executed on the network, we prevent actors from spamming the network. In order to prevent accidental or hostile infinite loops or other computational wastage in code, each transaction is required to set a limit to how many computational steps of code execution it can use. The fundamental unit of computation is "gas".
+[The London Upgrade](/history/#london) aims to make transacting on Ethereum more predictable for users by overhauling Ethereum's transaction-fee-mechanism. The high-level benefits introduced by this change include better transaction fee estimation, generally quicker transaction inclusion, and reducing the ETH rate of inflation by burning a percentage of transaction fees.
 
-Although a transaction includes a limit, any gas not used in a transaction is returned to the user.
+Starting with the London network upgrade, every block has a base fee, the minimum price per unit of gas for inclusion in this block, calculated by the network protocol based on network demand. Optionally, users can also set a priority fee (tip).
```

Comment:
"network protocol" seems redundant. You could use either, e.g. "by the protocol based on network demand", or "by the network based on demand for block space".

Reply:
> Optionally, users can also set a priority fee (tip).

The priority fee isn't optional. It is not strictly required, but we expect ~all transactions (except, say, MEV bundles) to have it. I would rephrase as: "Because this part of the transaction fee is burnt, users are also expected to set a priority fee (tip) in their transactions. This compensates miners for the work of executing and propagating user transactions in blocks, and is expected to be set automatically by most wallets.".